### PR TITLE
NDEV-2811 indexer: Mark Neon EVM transactions as corrupted if Neon logs were truncated

### PR DESCRIPTION
**Pull request by alfiedotwtf**
If we see the string "Log truncated" in the logs, we should mark transactions from that Solana block as corrupted

### CODE REVIEW

**Comment by alfiedotwtf** [comment]
Just a quick question about `proxy/indexer/indexed_objects.py`:

I haven't made any changes yet, but is `_finalize_log_list()` still correct in my PR, or should I add a change to now iterate over `self._done_neon_tx_list` first to check for ANY corrupted transactions, and then if there are:

    - short-circuit here by returning `None`
    - modify `_complete_neon_block()` to skip `add_neon_block(neon_block)`

File: proxy/indexer/indexed_objects.py
Comment (afalaleev) [inline proxy/indexer/indexed_objects.py:238]: You have a problem with restarts in this place.
The actual version doesn't have `is_log_truncated` property in the stored json.
As a result, on restart Indexer will fail if there are some stuck txs.

**Review by afalaleev — CHANGES_REQUESTED** [review]
> I haven't made any changes yet, but is _finalize_log_list() still correct in my PR...

Yes, it will be nice to refactor `_finalize_log_list()`.
It is critical to see all corrupted txs, not only first in a block

File: proxy/indexer/indexed_objects.py
Comment (afalaleev) [reply proxy/indexer/indexed_objects.py:238]: I made a mistake.
Everything is ok, `__init__()` has default value for `is_log_truncated`

**Review by afalaleev — DISMISSED** [review]

**Comment by alfiedotwtf** [comment]
> > I haven't made any changes yet, but is _finalize_log_list() still correct in my PR...
> 
> Yes, it will be nice to refactor `_finalize_log_list()`. It is critical to see all corrupted txs, not only first in a block

Ok, I'll commit my last changes to fix `_finalize_log_list()` now, and wait for approval one last time just in case before merging.

**Review by afalaleev — CHANGES_REQUESTED** [review]
Reject.
This code doesn't do anything.

**Comment by alfiedotwtf** [comment]
Maybe I have misunderstood, but just to make sure I've just stepped through it again.

Please see `proxy/indexer/indexed_objects.py : NeonIndexedBlockInfo::_finalize_log_list()`.

Here, `is_log_truncated = any(...)` will detect if there is a single truncated transaction log. If so, we loop through the list and `self._del_neon_tx(tx)` for every transaction. And because `elif tx.is_corrupted()` will return true, we should never reach `tx.complete_event_list()`.

In effect, this PR should now drain `self._neon_tx_dict` and return without adding any events to `neon_tx_res`.

**Comment by alfiedotwtf** [comment]
Closing as it will be implemented in the new proxy

**Pull request closed without merging**
